src/content/plugins/banner-plugin.md

```diff
@@ -28,9 +29,9 @@ new webpack.BannerPlugin(options);
   banner: string | function, // the banner as string or function, it will be wrapped in a comment
   raw: boolean, // if true, banner will not be wrapped in a comment
   entryOnly: boolean, // if true, the banner will only be added to the entry chunks
-  test: string | RegExp | Array,
-  include: string | RegExp | Array,
-  exclude: string | RegExp | Array,
+  test: string | RegExp | (string | RegExp)[],
+  include: string | RegExp | (string | RegExp)[],
+  exclude: string | RegExp | (string | RegExp)[],
```

Comment:
```suggestion
  test: string | RegExp | [string, RegExp],
  include: string | RegExp | [string, RegExp],
  exclude: string | RegExp | [srting, RegExp],
```

Reply:
Thanks for the link. But here's my reason, although both are valid in typescript/flow, `[string, RegExp]` means totally different thing from `(string | RegExp)[]`. Now that webpack is going to include [types for typescript](https://github.com/webpack/webpack/pull/10599), I think it's better to follow typescript's conventions.

Maybe we shall update the writers' guide?

Reply:
Interesting, would be great to match source types agaisnt the docs, automatically.

Reply:
Hi @chenxsan 
Its great that you suggest improvements, but currently to express an array of items limited to strings and RegExps we use `[string, RegExp]` in the documentation. We could continue the discussion of changing the types across the docs to be in TypeScript in the existing issue https://github.com/webpack/webpack.js.org/issues/3333